### ISSUE TITLE
Feature request: Adding folder where the .feature file resides automatically

### ISSUE THREAD

**Issue by mat013** [commercehub-oss/gradle-cucumber-jvm-plugin#16]
Hi

This request may seem too specific to how we work at my location, however I am going to give it a go anyway

I am thinking about organizing the test logic so that we only have a single test suite, which are structures as follows:

src/integration/java/feature1
src/integration/java/feature1/steps
src/integration/java/feature1/steps/StepsForFeature1.java
src/integration/java/feature1/somefeature11.feature
src/integration/java/feature1/somefeature12.feature
src/integration/java/feature2
src/integration/java/feature2/steps
src/integration/java/feature2/steps/StepsForFeature2.java
src/integration/java/feature2/somefeature21.feature
src/integration/java/feature2/somefeature22.feature

Supposedly you want to have StepsForFeature1 steps to be restricted .feature files in feature1 and StepsForFeature2 is restricted to .feature files in feature2. This can be done by maintaining the list of glue files in stepDefinitionRoots list, and this needs to be maintained as well when adding a new feature. 

I would like to ask if we could automatically add folder that has .feature files are automatically added to the stepDefinitionRoot e.g. if we have feature1 folder containing .feature files feature1 is automatically added to the stepDefinitionRoots and so forth.

Let me know what you think or if it not understandable

**Comment by JayStGelais** [comment]
I'm not sure I understand exactly what you are asking for on this one. 

Though its probably worth mentioning that the underlying cucumber framework does not support organizing steps by feature and segregating them for matching. If you really need to separate step definitions, your best bet would be to create multiple isolated test suites.

**Comment by mat013** [comment]
Well I want certain steps defined for feature1 only be available for somefeature11 and somefeature12 and certain steps defined for feature2 only be available for somefeature21 and somefeature22. As far as I can see when I ran (and that may be coincidental) that if one test define that the stepDefinitionRoots = ['feature1'] then I was able to only make those steps available for somefeature11 and somefeature12 and so forth. However we need some logic to add ['feature'], and that is what I am suggesting.
